rusty_socket_core/src/dataframe.rs

```diff
@@ -58,6 +62,104 @@ impl DataFrame {
         
         payload_length_bits
     }
+
+    fn apply_mask(&mut self) {
+        if let Some(masking_key) = self.masking_key {
+            for (i, byte) in self.payload.iter_mut().enumerate() {
+                *byte ^= masking_key[i % 4];
+            }
+        }
+    }
+
+    fn unmask_payload(&mut self) {
+        self.apply_mask()
+    }
+}
+
+impl TryFrom<&[u8]> for DataFrame {
+    type Error = RsError;
+
+    fn try_from(data: &[u8]) -> Result<Self, Self::Error> {
+        if data.len() < 2 {
+            return Err(RsError::IncompleteData);
+        }
+
+        let fin_rscv_opcode = data[0];
+        let mask_payload_length = data[1];
+
+        if (fin_rscv_opcode >> 7) & 1 != 1 {
+            return Err(RsError::FragmentationNotSupported);
+        }
+        if (fin_rscv_opcode & 0b00001111) > 0xA {
```

Comment:
This does not handle the case when opcode is 3,4,5,6,7 which are reserved and not defined in the protocol.